app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -257,80 +275,104 @@ public void cameraPerm (){
-            Participant participant = FeelTripApplication.getParticipant();
-            Mood mood = new Mood(participant.getUserName());
-            if (showPublicOn) {
-                mood.setPublic();
+                if (inputMoodDescription.getText().toString().equals("")) {
+                    throw new NullPointerException();
+                } else {
+                    mood.setDescription(String.valueOf(inputMoodDescription.getText()));
+                }
+
+                mood.setDate(dateTime.getTime());
+
+                if (encodedPhoto != null) {
+                    mood.setImage(encodedPhoto);
+                } else {
+                    mood.setNullImage();
+                }
+                editMoodTask.execute(mood);
+                Log.d("tag", "Editing mood");
+                finish();
             } else {
-                mood.setPrivate();
-            }
-            if (locationOn) {
-                mood.setMapPosition(latitude, longitude);
-                // this is the setter for latitude and longitude
-            }
-            else {
-                mood.setNullLocation();
-            }
-            mood.setEmotionalState(emotionalState);
-            mood.setSocialSit(socialSit);
-            mood.setDescription(String.valueOf(inputMoodDescription.getText())); //TODO: append the emotionalState upon fetching from Elasticsearch
-            mood.setDate(dateTime.getTime());
+                ElasticSearchController.AddMoodTask addMoodTask = new ElasticSearchController.AddMoodTask(this);
+                Participant participant = FeelTripApplication.getParticipant();
+                if (showPublicOn) {
+                    mood.setPublic();
+                } else {
+                    mood.setPrivate();
+                }
+                if (locationOn) {
+                    mood.setMapPosition(latitude, longitude);
+                    // this is the setter for latitude and longitude
+                } else {
+                    mood.setNullLocation();
+                }
+                if (emotionalState.equals("")) {
+                    throw new RuntimeException();
+                } else {
+                    mood.setEmotionalState(emotionalState);
+                }
 
-            if(encodedPhoto != null) {
-                mood.setImage(encodedPhoto);
-            }
-            else {
-                mood.setNullImage();
-            }
+                mood.setEmoji(emojiID);
+
+                mood.setSocialSit(socialSit);
+
+                if (inputMoodDescription.getText().toString().equals("")) {
+                    throw new NullPointerException();
+                } else {
+                    mood.setDescription(String.valueOf(inputMoodDescription.getText()));
+                }
+                mood.setDate(dateTime.getTime());
+
+                if (encodedPhoto != null) {
+                    mood.setImage(encodedPhoto);
+                } else {
```

Comment:
The thing about the setImage() function, along with all of the other functions, is it will only update the field of the mood if it detects an actual change has been made. So even though we "call" it twice, that failsafe ensures that we only actually call it once, so runtime remains about the same.